packages/chatbot-server-mongodb-public/src/test/evalHelpers.ts

```diff
@@ -25,3 +31,28 @@ export const openAiClient = new AzureOpenAI({
   endpoint: OPENAI_ENDPOINT,
   apiVersion: OPENAI_API_VERSION,
 });
+
+export function getLastUserMessageFromMessages(
+  messages: SomeMessage[]
+): UserMessage {
+  const userMessage = [...messages].reverse().find((m) => m.role === "user");
+  assert(userMessage, "Conversation must have a UserMessage");
+  return userMessage as UserMessage;
+}
+export function getLastAssistantMessageFromMessages(
+  messages: SomeMessage[]
+): AssistantMessage {
+  const assistantMessage = [...messages]
+    .reverse()
+    .find((m) => m.role === "assistant");
+  assert(assistantMessage, "Conversation must have a AssistantMessage");
+  return assistantMessage as AssistantMessage;
+}
+
+export function getContextsFromUserMessage(userMessage: UserMessage) {
+  const { data: contexts } = z
+    .array(z.string())
+    .safeParse(userMessage.contextContent?.map((cc) => cc.text));
+  // Return empty array if no context text found
+  return contexts ?? [];
+}
```

Comment:
Should we have tests for these helpers?

Reply:
i dont think we need b/c they're just used in the evals.

can add if you think important